packages/es-dev-server/src/utils/MessageStream.ts

```diff
@@ -0,0 +1,48 @@
+import { SSEStream } from './utils';
+import { EventEmitter } from 'events';
+import { Socket } from 'net';
+import { Context } from 'koa';
+
+export class MessageStream extends EventEmitter {
+  private __stream = new SSEStream();
+
+  constructor(public referer: string, private __socket: Socket, private __context: Context) {
```

Comment:
im confused, shouldnt these args still be set on `this`?

Reply:
This is typescript shorthand, when you add private/public/protected in front of the constructor parameter it's marked as a class field.

Reply:
Cool, TIL